crates/polars-arrow/src/bit_util.rs

```diff
@@ -79,3 +81,93 @@ pub fn ceil(value: usize, divisor: usize) -> usize {
         quot
     }
 }
+
+#[inline]
+fn get_zeroes(chunk: u64) -> u32 {
+    if cfg!(target_endian = "little") {
+        chunk.trailing_zeros()
+    } else {
+        chunk.leading_zeros()
+    }
+}
+
+#[inline]
+fn get_ones(chunk: u64) -> u32 {
+    if cfg!(target_endian = "little") {
+        chunk.trailing_ones()
+    } else {
+        chunk.leading_ones()
+    }
+}
+
+fn first_set_bit_impl<I>(mut mask_chunks: I) -> usize
+where
+    I: BitChunkIterExact<u64>,
+{
+    let mut total = 0usize;
+    let size = 64;
+    for chunk in &mut mask_chunks {
+        let pos = get_zeroes(chunk);
+        if pos != size {
+            return total + pos as usize;
+        } else {
+            total += size as usize
+        }
+    }
+    if let Some(pos) = mask_chunks.remainder_iter().position(|v| v) {
+        total += pos;
+        return total;
+    }
+    // all null, return the first
+    0
+}
+
+pub fn first_set_bit(mask: &Bitmap) -> usize {
```

Comment:
I don't like this implementation, I think it could be a lot faster using `[u8]::align_to::<u64>`. Could be even faster with SIMD.

Reply:
This two method is just moved from `arg_min_max.rs`, I didn't delve into the details of code.  

>I think it could be a lot faster using [u8]::align_to::<u64>. Could be even faster with SIMD.

It seems possible, but I still need to think carefully. I tend not to put this optimization to this PR because we have not changed the original code logic here. WDYT?

Reply:
Did you move or copy it from the original location? I don't think we should duplicate this method so we can have a single good implementation.